pkg/agent/agent_linux.go

```diff
@@ -17,23 +17,164 @@
+	uplinkNetConfig.MAC = adapter.HardwareAddr
+	uplinkNetConfig.IP = i.nodeConfig.NodeIPAddr
+	uplinkNetConfig.Index = adapter.Index
+	// Gateway and DNSServers are not configured at adapter in Linux
+	// Limitation: dynamic DNS servers will be lost after DHCP lease expired
+	uplinkNetConfig.Gateway = ""
+	uplinkNetConfig.DNSServers = ""
+	// Save routes which are configured on the uplink interface.
+	// The routes on the host will be lost when moving the network configuration of the uplink interface
+	// to the OVS bridge local interface. The saved routes will be restored on host after that.
+	if err = i.saveHostRoutes(); err != nil {
+		return err
+	}
+
+	// Set datapathID of OVS bridge.
+	// If no datapathID configured explicitly, the reconfiguration operation will change OVS bridge datapathID
+	// and break the OpenFlow channel.
+	// The length of datapathID is 64 bits, the lower 48-bits are for a MAC address, while the upper 16-bits are
+	// implementer-defined. Antrea uses "0x0000" for the upper 16-bits.
+	datapathID := strings.Replace(uplinkNetConfig.MAC.String(), ":", "", -1)
+	datapathID = "0000" + datapathID
+	if err = i.ovsBridgeClient.SetDatapathID(datapathID); err != nil {
+		klog.Errorf("Failed to set datapath_id %s: %v", datapathID, err)
+		return err
+	}
+
+	// Create local port.
+	brName := i.ovsBridgeClient.GetBridgeName()
+	if _, err = i.ovsBridgeClient.GetOFPort(brName); err == nil {
+		klog.Infof("OVS bridge local port %s already exists, skip the configuration", brName)
+		// If uplink is internal port, get the real uplink interface.
+		// This branch is used when antrea-agent get a hard restart (e.g. SIGKILL)
+		if uplinkNetConfig.Name == brName {
+			ports, err := i.ovsBridgeClient.GetPortList()
+			if err != nil {
+				klog.Errorf("Cannot list OVS ports.")
+				return err
+			}
+			klog.Infof("Found ports from OVS bridge: %+v", ports)
+			var uplinkPort *ovsconfig.OVSPortData
+			for index := range ports {
+				if ports[index].OFPort == config.UplinkOFPort {
+					uplinkPort = &ports[index]
+					break
+				}
+			}
+			if uplinkPort == nil {
+				return fmt.Errorf("Cannot find uplink port from OVS bridge %s", brName)
+			}
+			adapter, err2 := net.InterfaceByName(uplinkPort.Name)
+			if err2 != nil {
+				klog.Errorf("Cannot find uplink port %s", uplinkPort.Name)
+				return err2
+			}
+			klog.Infof("Found uplink device %s", adapter.Name)
+			uplinkNetConfig.Name = adapter.Name
+			uplinkNetConfig.Index = adapter.Index
+		}
+	} else {
+		// OVS does not receive "ofport_request" param when creating local port, so here use config.AutoAssignedOFPort=0
+		// to ignore this param.
+		if _, err = i.ovsBridgeClient.CreateInternalPort(brName, config.AutoAssignedOFPort, nil); err != nil {
+			return err
+		}
+	}
+
+	// If uplink is already exists, return.
+	uplink := uplinkNetConfig.Name
+	if _, err := i.ovsBridgeClient.GetOFPort(uplink); err == nil {
+		klog.Infof("Uplink %s already exists, skip the configuration", uplink)
```

Comment:
Maybe I misunderstood the workflow, but in this case we're not adding the interface to ifaceStore. Not sure if this is a problem

Reply:
Added cache here.